pkg/service/service.go

```diff
@@ -108,7 +109,8 @@ func Apply(
 			logger.Errorf("error updating status of service %s: %v",
 				entity.ID, uerr)
 		}
-	}()
+	}
+	defer recycle()
```

Comment:
This defer seems useless, since recycle is already called in gopool.Go.

Could you change the name recycle, since no recycle action here?